src/track/beatutils.cpp

```diff
@@ -12,37 +12,314 @@
+            continue;
+        } else {
+            // This may not be case when our median window is even, we can't use it
+            // Because find() will return an iterator pointing to end that we will *
+            if (tempoFrequency.contains(newStableTempo)) {
+                lastBeatChange = currentBeat - tempoMedianFilter.lag() - mostFrequentTempo.lag();
+                stableTemposByPosition[lastBeatChange] = newStableTempo;
+            }
+        }
+    }
+    stableTemposByPosition[tempoList.count()] = medianBpm;
+    return stableTemposByPosition;
+}
+
+void BeatUtils::RemoveSmallArrhythmic(
+        QVector<double>& rawBeats,  int sampleRate, QMap<int, double> &stableTemposByPosition) {
+    // A common problem the analyzer has is to detect arrhythmic regions
+    // of tacks with a constant tempo as in a different unsteady tempo. 
+    // This happens frequently on builds and breaks with heavy effects on edm music.
+    // Since these occurs most on beatless regions we do not want them to be
+    // on a different tempo, because they are still syncable in the true tempo
+    // We arbitraly remove these arrhythmic regions if they are short than 16s
+    auto positionsWithTempoChange = stableTemposByPosition.keys();
+    auto tempoValues = stableTemposByPosition.values();
+    QVector<double> anchoredBeats;
+    anchoredBeats.reserve(rawBeats.size());
+    anchoredBeats << QVector<double>::fromStdVector(std::vector<double>(
+                rawBeats.begin(), rawBeats.begin() + positionsWithTempoChange[1]));
+
+    for (int i = 2; i < positionsWithTempoChange.size(); i+=1) {
+        int smallInBeats =  16 / (60 / tempoValues[i-1]);
+        int limitAtLeft = positionsWithTempoChange[i-1];
+        int limitAtRight = positionsWithTempoChange[i];
+        int lenghtOfChange = limitAtRight - limitAtLeft;
+        double previousTempo = (stableTemposByPosition.lowerBound(limitAtLeft)-1).value();
+
+        if (lenghtOfChange <= smallInBeats) {
+            stableTemposByPosition.remove(limitAtLeft);
+            double beatOffset = rawBeats[limitAtLeft];
+            double beatLength = floor(((60.0 * sampleRate) / previousTempo) + 0.5);
+            while (beatOffset < rawBeats[limitAtRight]) {
+                anchoredBeats << beatOffset;
+                beatOffset += beatLength;
+            }
+
+        } else {
+            anchoredBeats << QVector<double>::fromStdVector(std::vector<double>(
+                    rawBeats.begin() + positionsWithTempoChange[i-1],
+                    rawBeats.begin() + positionsWithTempoChange[i]));
+        }
+    }
+    // We may have shinkred or enlarged our beat vector so we make sure our  
+    // last change sentinal does happens in the right place.
+    stableTemposByPosition.remove(stableTemposByPosition.last());
+    stableTemposByPosition[anchoredBeats.size() -1] = stableTemposByPosition[0];
+    rawBeats = anchoredBeats;
```

Comment:
Isn't this a unnecessary deep copy? 

I guess it shool be possible to edit the beats on the fly.

Reply:
We need to copy either way, because by editing them on the fly we may need to shift almost the whole vector that will be copying in the same way, right?
Then this makes it the code easier at a reasonable cost? Because then we are also iterating on the same vector we are editing and things can get messy easily

Reply:
Ah Ok, I Understand. 
Please describe this as code comment.

Reply:
done